src/test/index/TestIndexCompressV3.cpp

```diff
@@ -0,0 +1,325 @@
+
+                _CLDECDELETE(term);
+            }
+        }
+        _CLFINALLY({
+            _CLDECDELETE(term);
+            enumerator->close();
+            _CLDELETE(enumerator);
+        })
+
+    } catch (...) {
+        eptr = std::current_exception();
+    }
+    FINALLY(eptr, {
+        reader->close();
+        _CLLDELETE(reader);
+    })
+}
+
+static void index_compaction(RAMDirectory* tmp_dir, std::vector<lucene::store::Directory*> srcDirs,
+                             std::vector<lucene::store::Directory*> destDirs, int32_t count) {
+    auto* analyzer = _CLNEW lucene::analysis::SimpleAnalyzer<char>;
+    auto* indexwriter = _CLNEW lucene::index::IndexWriter(tmp_dir, analyzer, true);
+
+    std::vector<std::vector<std::pair<uint32_t, uint32_t>>> trans_vec(
+            srcDirs.size(), std::vector<std::pair<uint32_t, uint32_t>>(count));
+    int32_t idx = 0;
+    int32_t id = 0;
+    for (int32_t i = 0; i < count; i++) {
+        for (int32_t j = 0; j < srcDirs.size(); j++) {
+            if (id == count * destDirs.size()) {
+                idx++;
+                id = 0;
+            }
+            trans_vec[j][i] = std::make_pair(idx, id++);
+        }
+    }
+
+    std::vector<uint32_t> dest_index_docs(destDirs.size());
+    for (int32_t i = 0; i < destDirs.size(); i++) {
+        dest_index_docs[i] = count * destDirs.size();
+    }
+
+    std::exception_ptr eptr;
+    try {
+        indexwriter->indexCompaction(srcDirs, destDirs, trans_vec, dest_index_docs);
+    } catch (...) {
+        eptr = std::current_exception();
+    }
+    FINALLY(eptr, {
+        indexwriter->close();
+        _CLDELETE(indexwriter);
+        _CLDELETE(analyzer);
+    })
+}
+
+void TestIndexByteArray(CuTest* tc) {
+    RAMDirectory dir;
+    auto ram_out = dir.createOutput("TestIndexByteArray");
+
+    v2::GrowableByteArrayDataOutput out;
+    for (int32_t i = 0; i < doc_count; i++) {
+        out.writeVInt(i);
+    }
+    out.writeCompressedTo(ram_out);
+    ram_out->close();
+
+    IndexInput* ram_in = nullptr;
+    CLuceneError error;
+    bool ret = dir.openInput("TestIndexByteArray", ram_in, error);
+    if (!ret) {
+        std::cout << error.what() << std::endl;
+    }
+    assertTrue(ret);
+
+    v2::ByteArrayDataInput in;
+    in.readCompressedFrom(ram_in);
+    for (int32_t i = 0; i < doc_count; i++) {
+        assertEquals(in.readVInt(), i);
+    }
+
+    _CLDELETE(ram_out);
+    _CLDELETE(ram_in);
+
+    std::cout << "\nTestIndexByteArray sucess" << std::endl;
+}
+
+void TestIndexCompressV3(CuTest* tc) {
+    std::srand(getDaySeed());
+
+    std::string name = "v2_field_name";
+    std::vector<std::string> datas;
+    for (int32_t i = 0; i < doc_count; i++) {
+        std::string ip_v4 = generateRandomIP();
+        datas.emplace_back(ip_v4);
+    }
+
+    RAMDirectory dir;
+    write_index(name, &dir, IndexVersion::kV3, false, datas);
+
+    try {
+        read_index(&dir, doc_count);
```

Comment:
read does not check any data.

Reply:
In the read_index function, the position range is checked, and an exception is thrown if there is an error.